create.ps1

```diff
@@ -1,258 +1,449 @@
+                Method      = "PATCH"
+                Body        = ($updateAccountBody | ConvertTo-Json -Depth 10)
+                ContentType = 'application/json; charset=utf-8'
+                Verbose     = $false
+                ErrorAction = "Stop"
+            }
 
-        # Force the account disabled
-        $Body.accountEnabled = $False
+            Write-Verbose "SplatParams: $($updateAccountSplatParams | ConvertTo-Json)"
 
-        $SplatParams = @{
-            Uri    = "$($actionContext.Configuration.BaseUrl)/Users/$($outputContext.AccountReference)/bulk"
-            Method = "Patch"
-            Body   = $Body
-        }
+            if (-Not($actionContext.DryRun -eq $true)) {
+                # Add header after printing splat
+                $updateAccountSplatParams['Headers'] = $headers
 
-        if (-Not ($actionContext.DryRun -eq $True)) {
-            [void] (Invoke-RestMethod @LisaRequest @SplatParams)
-        }
-        else {
-            Write-Verbose -Verbose ($Body | ConvertTo-Json)
-        }
+                $updateAccountResponse = Invoke-RestMethod @updateAccountSplatParams
 
-        $outputContext.AuditLogs.Add([PSCustomObject]@{
-                Action  = "CreateAccount" # Optionally specify a different action for this audit log
-                Message = "Created account for '$($personContext.Person.DisplayName)'. Id: $($outputContext.AccountReference)"
-                IsError = $False
-            })
-
-        # Set the manager
-        if ($actionContext.References.ManagerAccount) {
-            $SplatParams = @{
-                Uri    = "$($actionContext.Configuration.BaseUrl)/Users/$($outputContext.AccountReference)/Manager"
-                Method = "Put"
-                Body   = $actionContext.References.ManagerAccount
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "Updated created account with AccountReference: $($outputContext.AccountReference | ConvertTo-Json). Updated properties [$($updateAccountBody | ConvertTo-Json)."
+                        IsError = $false
+                    })
+            }
+            else {
+                Write-Warning "DryRun: Would update created account. Updated properties [$($updateAccountBody | ConvertTo-Json)."
             }
+            #endregion Update account
+
+            #region Manager
+            if ($actionContext.Configuration.setManagerOnCreate -eq $true) {
+                #region Calulate manager action
+                if (-not[String]::IsNullOrEmpty(($actionContext.References.ManagerAccount))) {
+                    $actionManager = "Set"
+                    $currentKPNLisaManagerAccountId = $actionContext.References.ManagerAccount
+                }
+                else {
+                    $actionManager = "AccountReferenceEmpty"
+                }
```

Comment:
Deze validatie zouden we mogelijk beter op een hoger niveau kunnen plaatsen. Bv. direct na de `GET` account. Dat maakt het proces logischer. Eerst valideren / controleren wat we moeten doen en pas daarna allemaal 'onder elkaar!' uitvoeren.

Reply:
In theorie, ja, nu is dit deel later toegevoegd en als los onderdeel dus erbij gezet.
Zo'n aanpassing verandert niks in de werking van de connector, dus dit laat ik voor nu zo.